include/nbl/asset/IRenderpass.h

```diff
@@ -0,0 +1,234 @@
+    EPSF_TRANSFORM_FEEDBACK_BIT_EXT = 0x01000000,
+    EPSF_CONDITIONAL_RENDERING_BIT_EXT = 0x00040000,
+    EPSF_RAY_TRACING_SHADER_BIT_KHR = 0x00200000,
+    EPSF_ACCELERATION_STRUCTURE_BUILD_BIT_KHR = 0x02000000,
+    EPSF_SHADING_RATE_IMAGE_BIT_NV = 0x00400000,
+    EPSF_TASK_SHADER_BIT_NV = 0x00080000,
+    EPSF_MESH_SHADER_BIT_NV = 0x00100000,
+    EPSF_FRAGMENT_DENSITY_PROCESS_BIT_EXT = 0x00800000,
+    EPSF_COMMAND_PREPROCESS_BIT_NV = 0x00020000
+};
+// TODO probably should be moved to separate header
+enum E_ACCESS_FLAGS : uint32_t
+{
+    EAF_INDIRECT_COMMAND_READ_BIT = 0x00000001,
+    EAF_INDEX_READ_BIT = 0x00000002,
+    EAF_VERTEX_ATTRIBUTE_READ_BIT = 0x00000004,
+    EAF_UNIFORM_READ_BIT = 0x00000008,
+    EAF_INPUT_ATTACHMENT_READ_BIT = 0x00000010,
+    EAF_SHADER_READ_BIT = 0x00000020,
+    EAF_SHADER_WRITE_BIT = 0x00000040,
+    EAF_COLOR_ATTACHMENT_READ_BIT = 0x00000080,
+    EAF_COLOR_ATTACHMENT_WRITE_BIT = 0x00000100,
+    EAF_DEPTH_STENCIL_ATTACHMENT_READ_BIT = 0x00000200,
+    EAF_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT = 0x00000400,
+    EAF_TRANSFER_READ_BIT = 0x00000800,
+    EAF_TRANSFER_WRITE_BIT = 0x00001000,
+    EAF_HOST_READ_BIT = 0x00002000,
+    EAF_HOST_WRITE_BIT = 0x00004000,
+    EAF_MEMORY_READ_BIT = 0x00008000,
+    EAF_MEMORY_WRITE_BIT = 0x00010000,
+    EAF_TRANSFORM_FEEDBACK_WRITE_BIT_EXT = 0x02000000,
+    EAF_TRANSFORM_FEEDBACK_COUNTER_READ_BIT_EXT = 0x04000000,
+    EAF_TRANSFORM_FEEDBACK_COUNTER_WRITE_BIT_EXT = 0x08000000,
+    EAF_CONDITIONAL_RENDERING_READ_BIT_EXT = 0x00100000,
+    EAF_COLOR_ATTACHMENT_READ_NONCOHERENT_BIT_EXT = 0x00080000,
+    EAF_ACCELERATION_STRUCTURE_READ_BIT_KHR = 0x00200000,
+    EAF_ACCELERATION_STRUCTURE_WRITE_BIT_KHR = 0x00400000,
+    EAF_SHADING_RATE_IMAGE_READ_BIT_NV = 0x00800000,
+    EAF_FRAGMENT_DENSITY_MAP_READ_BIT_EXT = 0x01000000,
+    EAF_COMMAND_PREPROCESS_READ_BIT_NV = 0x00020000,
+    EAF_COMMAND_PREPROCESS_WRITE_BIT_NV = 0x00040000
+};
+// TODO probably should be moved to separate header
+enum E_DEPENDENCY_FLAGS
+{
+    EDF_BY_REGION_BIT = 0x01,
+    EDF_VIEW_LOCAL_BIT = 0x02,
+    EDF_DEVICE_GROUP_BIT = 0x04
+};
+
+class IRenderpass
+{
+public:
+    enum E_LOAD_OP : uint8_t
+    {
+        ELO_LOAD,
+        ELO_CLEAR,
+        ELO_DONT_CARE
+    };
+    enum E_STORE_OP : uint8_t
+    {
+        ESO_STORE,
+        ESO_DONT_CARE
+    };
+    enum E_SUBPASS_DESCRIPTION_FLAGS
+    {
+        ESDF_NONE = 0x00,
+        ESDF_PER_VIEW_ATTRIBUTES_BIT = 0x01,
+        ESDF_PER_VIEW_POSITION_X_ONLY_BIT = 0x02
+    };
+    enum E_ATTACHMENT_POINT : uint8_t
+    {
+        EAP_DEPTH,
+        EAP_STENCIL,
+        EAP_COLOR_0,
+        EAP_INPUT_0,
+        EAP_COLOR_1,
+        EAP_INPUT_1,
+        EAP_COLOR_2,
+        EAP_INPUT_2,
+        EAP_COLOR_3,
+        EAP_INPUT_3,
+        EAP_COLOR_4,
+        EAP_INPUT_4,
+        EAP_COLOR_5,
+        EAP_INPUT_5,
+        EAP_COLOR_6,
+        EAP_INPUT_6,
+        EAP_COLOR_7,
+        EAP_INPUT_7,
```

Comment:
we should probably have all the COLOR first and then just a symbolic `EAP_INPUT_0` and then no `EAP_COUNT`... then we could have arbitrary number of input attachments

Reply:
(or a few input enums and then a count enum)

Reply:
(yes I know its my own design I'm criticising)

Reply:
so what happened in the end?

Reply:
I repeat my question

Reply:
it's very vulkan-like, renderpass creation just takes a pointer (+size uint) to array of attachment descriptions